core/src/main/java/com/linecorp/armeria/unsafe/ByteBufHttpData.java

```diff
@@ -20,30 +20,53 @@
 
 import java.io.InputStream;
 import java.nio.charset.Charset;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
 
 import com.google.common.base.MoreObjects;
 
 import com.linecorp.armeria.common.AbstractHttpData;
 import com.linecorp.armeria.common.HttpData;
-import com.linecorp.armeria.common.util.UnstableApi;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.ByteBufHolder;
 import io.netty.buffer.ByteBufInputStream;
 import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.EmptyByteBuf;
 import io.netty.buffer.Unpooled;
+import io.netty.buffer.UnpooledByteBufAllocator;
 
 /**
  * An {@link HttpData} that is backed by a {@link ByteBuf} for optimizing certain internal use cases. Not for
  * general use.
+ *
+ * @deprecated Use {@link PooledHttpData}.
  */
-@UnstableApi
-public final class ByteBufHttpData extends AbstractHttpData implements ByteBufHolder {
+@Deprecated
+public final class ByteBufHttpData extends AbstractHttpData implements PooledHttpData {
+
+    private static final AtomicIntegerFieldUpdater<ByteBufHttpData>
+            closedUpdater = AtomicIntegerFieldUpdater.newUpdater(ByteBufHttpData.class, "closed");
+
+    static final ByteBufHttpData EMPTY = new ByteBufHttpData(
+            new EmptyByteBuf(UnpooledByteBufAllocator.DEFAULT), false);
+
+    /**
+     * Converts non-pooled {@link HttpData} into {@link PooledHttpData}.
+     */
+    static PooledHttpData convert(HttpData data) {
```

Comment:
I think `from` is more suitable for our API style.
```suggestion
    static PooledHttpData from(HttpData data) {
```